pkg/sds/provider/vault/identity_cert_provider.go

```diff
@@ -0,0 +1,63 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	vault "github.com/hashicorp/vault/api"
+
+	"github.com/Kong/kuma/pkg/sds/auth"
+	"github.com/Kong/kuma/pkg/sds/provider"
+	"github.com/Kong/kuma/pkg/sds/provider/identity"
+)
+
+func NewIdentityCertProvider(client *vault.Client) provider.SecretProvider {
+	return &identityCertProvider{
+		client: client,
+	}
+}
+
+type identityCertProvider struct {
+	client *vault.Client
+}
+
+var _ provider.SecretProvider = &identityCertProvider{}
+
+func (m *identityCertProvider) Get(ctx context.Context, name string, requestor auth.Identity) (provider.Secret, error) {
+	data := map[string]interface{}{
+		"ttl":      "720h",
```

Comment:
Why `ttl` is so big? Is there any rotation mechanism?

Reply:
Is 30 days long though?
No, right now there is no rotation mechanism yet.

For workload certs from builtin/provided CA we've got 90 days `DefaultWorkloadCertValidityPeriod = 90 * 24 * time.Hour`